File: Lynda Courses/iOS App Development Essential Training with Simon Allardice/TableDemo/TableDemo/ViewController.m
//
//  ViewController.m
//  TableDemo
//
//  Created by Eude K Lesperance on 10/10/14.
//  Copyright (c) 2014 Eude K Lesperance. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

-(NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
    return 5;
}

-(UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath{
    UITableViewCell *cell =[[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"cell"];
    
    cell.textLabel.text = @"This is a cell";
    return cell;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: TopCoder/SurveyApp/SurveyApp/ViewController.swift
//
//  ViewController.swift
//  SurveyApp
//
//  Created by Eude Kinsley Lesperance on 10/26/15.
//  Copyright © 2015 Learning. All rights reserved.
//

import UIKit

class ViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {
    
    @IBOutlet weak var surveyTableView: UITableView!
    
    
    var data: [NSDictionary] = [];
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        self.surveyTableView.registerClass(UITableViewCell.self, forCellReuseIdentifier: "cell")
        
        let dataPath = NSBundle.mainBundle().pathForResource("data", ofType: "plist")
        data = NSArray(contentsOfFile: dataPath!)! as! [NSDictionary]
    }
    
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return data.count;
    }
    
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCellWithIdentifier("cell", forIndexPath: indexPath) 
        
        cell.textLabel?.text = data[indexPath.row]["title"] as? String
        return cell
    }


}

